extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java

```diff
@@ -260,9 +260,13 @@ <T> T makePostRequest(String url, Optional<Map<String, String>> parameters,
    * Note that making this a separate method to avoid polluting throw lists.
    */
   private void maybeRethrowAsUploadError(HttpResponseException e) throws UploadErrorException {
-    if (e.getStatusCode() == 400 && e.getContent()
-        .contains("Checksum from header does not match received payload content.")) {
-      throw new UploadErrorException(ERROR_HASH_MISMATCH, e);
+    if (e.getStatusCode() == 400) {
```

Comment:
optional nit: your choice, but maybe this is clearer as a single expression? `e.getStatusCode() == 400 && (e.getContent().contains(GOOG_...) || e.getContent().contains(GOOG_...))`

I think this makes your comment about delegation a bit clearer (really _everything_ will be delegated to be handled eleswhere - even the non-400s).

Reply:
Thanks! I feel like grouping those error messages in one line doesn't make much difference. Actually, how about I update the coment to explicitly mention other 400 errors and non-400 errors will be delegated?